buildSrc/src/main/kotlin/DokkaExts.kt

```diff
@@ -186,7 +202,7 @@ fun Project.dokkaKotlinJar(): TaskProvider<Jar> = tasks.getOrCreate("dokkaKotlin
  */
 fun AbstractDokkaTask.isInPublishingGraph(): Boolean =
     project.gradle.taskGraph.allTasks.any {
-        it.name == "publish"
+        it.name == "publish" || it.name.contains("dokkaGenerate")
```

Comment:
Two points here:

1. It does not look like `publishToMavenLocal` is excluded as stated in docs.
2. Please mention in docs that `dokkaGenerate` is a subtask of publishing (as I understand).